types/puppeteer/index.d.ts

```diff
@@ -346,90 +729,364 @@ export interface FrameBase {
+   * @param cookies The cookies to set.
+   */
   setCookie(...cookies: SetCookie[]): Promise<void>;
+
+  /**
+   * The extra HTTP headers will be sent with every request the page initiates.
+   * @param headers An object containing additional http headers to be sent with every request. All header values must be strings.
+   */
   setExtraHTTPHeaders(headers: Headers): Promise<void>;
+
+  /**
+   * Determines whether JavaScript is enabled on the page.
+   * @param enable Whether or not to enable JavaScript on the page.
+   */
   setJavaScriptEnabled(enabled: boolean): Promise<void>;
-  setRequestInterceptionEnabled(value: boolean): Promise<void>;
+
+  /**
+   * Determines whether the offline mode is enabled.
+   * @param enabled When `true`, enables the offline mode for the page.
+   */
+  setOfflineMode(enabled: boolean): Promise<void>;
+
+  /**
+   * Determines whether the request interception is enabled.
+   * @param enabled When `true` the methods `request.abort`, `request.continue` and `request.respond` must be used.
+   */
+  setRequestInterception(enabled: boolean): Promise<void>;
+
+  /**
+   * Specifies the User-Agent used in this page.
+   * @param userAgent The user-agent to be used in the page.
+   */
   setUserAgent(userAgent: string): Promise<void>;
+  /**
+   * Sets the viewport of the page.
+   * @param viewport The viewport parameters.
+   */
   setViewport(viewport: Viewport): Promise<void>;
+
+  /**
+   * This method fetches an element with `selector`, scrolls it into view if needed,
+   * and then uses page.touchscreen to tap in the center of the element.
+   * @param selector A `selector` to search for element to tap. If there are multiple elements
+   * satisfying the selector, the first will be tapped.
+   */
   tap(selector: string): Promise<void>;
+
+  /** Returns the page's title. */
+  title(): Promise<string>;
+
+  /** Returns the virtual touchscreen object. */
   touchscreen: Touchscreen;
+
+  /** Returns the tracing object. */
   tracing: Tracing;
+
+  /**
+   * Sends a `keydown`, `keypress/input`, and `keyup` event for each character in the text.
+   * @param selector A selector of an element to type into. If there are multiple elements satisfying the selector, the first will be used.
+   * @param text: A text to type into a focused element.
+   * @param options: The typing parameters.
+   */
   type(selector: string, text: string, options?: { delay: number }): Promise<void>;
+
+  /**
+   * The page's URL. This is a shortcut for `page.mainFrame().url()`
+   */
+  url(): string;
+
+  /** Gets the page viewport. */
   viewport(): Viewport;
+
+  /**
+   * Wait for the page navigation occur.
+   * @param options The navigation parameters.
+   */
   waitForNavigation(options?: NavigationOptions): Promise<Response>;
 }
 
+/** A Browser is created when Puppeteer connects to a Chromium instance, either through puppeteer.launch or puppeteer.connect. */
 export interface Browser {
+  on<K extends keyof BrowserEventObj>(
```

Comment:
The [docs](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-browser) don't mention an `on()`-method. Perhaps their document is outdated, or this doesn't exist.

Reply:
The docs mention the events (disconnected, targetchanged, targetcreated and the targetdestroyed). The `Browser` class [extends](https://github.com/GoogleChrome/puppeteer/blob/master/lib/Browser.js#L21) the [EventEmitter](https://nodejs.org/api/events.html) that has the `on` method.

I think we should be fine.

Reply:
You're right, thanks for clearing this up! Just had a look at the source and it indeed does extend event emitter. :radar: